api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +345,147 @@ type AzureWebhookReceiverConfig struct {
+type GenericWebhookReceiverConfig struct {
+	// SecretRef contains a reference to a Secret. For Project-scoped webhook
+	// receivers, the referenced Secret must be in the same namespace as the
+	// ProjectConfig.
+	//
+	// For cluster-scoped webhook receivers, the referenced Secret must be in the
+	// designated "cluster Secrets" namespace.
+	//
+	// The Secret's data map is expected to contain a `secret` key whose value is
+	// the shared secret used to authenticate the webhook requests sent by the
+	// generic webhook sender.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
+
+	// Actions is a list of actions to be performed when a webhook event is received.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Actions []GenericWebhookAction `json:"actions,omitempty" protobuf:"bytes,2,rep,name=actions"`
+}
+
+// GenericWebhookAction describes an action to be performed on a resource
+// and the conditions under which it should be performed.
+type GenericWebhookAction struct {
+	// Name is the name of the action to be performed.
+	//
+	// +kubebuilder:validation:Enum=Refresh;
+	Action GenericWebhookActionName `json:"action" protobuf:"bytes,1,opt,name=action"`
+
+	// MatchConditions is a list of conditions that must be met for the action to be performed.
+	//
+	// +optional
+	MatchConditions []ConditionSelector `json:"matchConditions,omitempty" protobuf:"bytes,2,rep,name=matchConditions"`
+
+	// Parameters contains additional parameters for the action.
+	//
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+
+	// Targets is a list of targets on which the action should be performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes a target resource for a generic webhook event.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource (e.g., "Warehouse", "Stage").
+	//
+	// +kubebuilder:validation:Required
+	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,2,opt,name=labelSelector"`
+
+	// IndexSelector is an index selector to identify the target resources.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,3,opt,name=indexSelector"`
```

Comment:
We need to make it clear that the criteria are the intersection (and) of what is selected by each of these.